src/parser.c

```diff
@@ -634,7 +634,10 @@ void read_numeric_param(block_t *parent, basic_block_t *bb, int is_neg)
             } while (is_hex(token[i]));
         } else { /* octal */
             do {
-                c = token[i++] - '0';
+                c = token[i++];
+                if (c > '7' || c < '0')
```

Comment:
The check for `c < '0'` is redundant because the loop will exit as soon as `is_digit()` in line 642 returns false.   
For example, consider the case `int a = 0.; funct(a);`, where the ASCII value of `'.'` is less than `'0'`.   
On the first iteration, `c = token[i++]` assigns `c` the value `'0'` and increments `i` to 1.    
Then, the check in the while loop at line 642 `is_digit(token[i = 1])` will become `is_digit('.')` and return false, causing the loop to exit.   
As a result, the check for `c < '0'` is never triggered when the comming digit is less then '0'.

Reply:
@fennecJ , yes, here doesn't need to check for `c < '0'`. Thank you for your feedback, I will improve it later.